From 11167ce99d350e90c47d53205a0e7c6c4afd1f5f Mon Sep 17 00:00:00 2001
From: Michel JAOUEN <michel.jaouen@st.com>
Date: Thu, 20 Feb 2025 14:20:13 +0100
Subject: [PATCH 1560/1695] plat: stm32mp2: set unsafe config to keep
 performance

After rebase on mbedtls 3.6.2, rsa key operation are degraded without
this flag

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git 7eb099e82045a32498b8bb03a251a0e81d655f39 plat: stm32mp2: set unsafe config to keep performance]
Change-Id: If95d33c3b389b0712a5273eb292ee0d13b05e67c
Signed-off-by: Michel JAOUEN <michel.jaouen@st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/438958
Domain-Review: Yann GAUTIER <yann.gautier@foss.st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Reviewed-by: Thomas BOURGOIN <thomas.bourgoin@foss.st.com>
---
 core/arch/arm/plat-stm32mp1/conf.mk | 3 +++
 core/arch/arm/plat-stm32mp2/conf.mk | 3 +++
 2 files changed, 6 insertions(+)

diff --git a/core/arch/arm/plat-stm32mp1/conf.mk b/core/arch/arm/plat-stm32mp1/conf.mk
index 69536d317..c1190aa8e 100644
--- a/core/arch/arm/plat-stm32mp1/conf.mk
+++ b/core/arch/arm/plat-stm32mp1/conf.mk
@@ -161,6 +161,9 @@ $(call force,CFG_STM32_SAES,n)
 CFG_WITH_SOFTWARE_PRNG ?= y
 endif
 
+# Force this mode to keep performance on RSA key operations
+CFG_CORE_UNSAFE_MODEXP ?= y
+
 # Default do not access external DT passed to non-secure boot stage
 CFG_EXTERNAL_DT ?= n
 
diff --git a/core/arch/arm/plat-stm32mp2/conf.mk b/core/arch/arm/plat-stm32mp2/conf.mk
index 36a652c74..a146d6a94 100644
--- a/core/arch/arm/plat-stm32mp2/conf.mk
+++ b/core/arch/arm/plat-stm32mp2/conf.mk
@@ -141,6 +141,9 @@ endif
 CFG_TZDRAM_START ?= 0x82000000
 CFG_TZDRAM_SIZE  ?= 0x02000000
 
+# Force this mode to keep performance on RSA key operations
+CFG_CORE_UNSAFE_MODEXP ?= y
+
 # Support DDR ranges up to 8GBytes (address range: 0x80000000 + DDR size)
 CFG_CORE_LARGE_PHYS_ADDR ?= y
 CFG_CORE_ARM64_PA_BITS ?= 34
-- 
2.47.2

